### PR TITLE
Add the api-test folder

### PR DESCRIPTION
**Pull request by elwoujdi**
This pull request proposes the transfer of the "api-test" folder from the finesse-backend repository to the api-test repository.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "editor.rulers": [80], // used for .md files
```
Comment (rngadam) [inline .vscode/settings.json:2]: not just yaml

File: README.md [hunk 2]
```diff
@@ -1,2 +1,72 @@
 # api-test
-This repository contains all the tools for testing and stressing the APIs of the AI Lab projects. 
+
+This repository contains all the tools for testing and stressing the APIs of the
```
Comment (rngadam) [inline README.md:3]: >80. please add .github Github Workflows

**Review by rngadam — CHANGES_REQUESTED** [review]
Please add relevant Github Workflows to repository to lint source before continuing.

File: .vscode/settings.json (hunk 1 above)
Comment (ibrahim-kabir) [reply .vscode/settings.json]: Done

File: README.md (hunk 2 above)
Comment (ibrahim-kabir) [reply README.md:3]: Done

**Comment by rngadam** [comment]
@ibrahim-kabir I see you've added the workflows but seems something went wrong as no check task gets triggered. Can you investigate why?

**Comment by SonOfLope** [comment]
@ibrahim-kabir did you try with act locally ? 
image

Your'e calling workflow-lint-test-python but dont have a tests folder. Also your'e making use of workflow-markdown-check with a specified config file '.mlc_config.json' which is absent in your files.

File: finesse/FINESSE_USAGE.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    correspond of being at the top of the list, and 0% would mean not in the
+    list.
+- **Round trip time**
+  - Measure round trip time of each request
+- **Summary statistical value**
+  - Measure the average, median, standard deviation, minimum and maximal accuracy scores and round trip time
+
+## Diagram
+
+![Alt text](diagram.png)
```
Comment (rngadam) [inline finesse/FINESSE_USAGE.md:50]: where's the source of this diagram? Mermaid? need the source version-controlled.

File: finesse/FINESSE_USAGE.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+# Design of the Finesse Benchmark Tool
```
Comment (rngadam) [inline finesse/FINESSE_USAGE.md:1]: we want to generalize to backend API testing, not just Finesse. review documentation to update.

File: finesse/accuracy_functions.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+from collections import namedtuple
+
+OUTPUT_FOLDER = "./finesse/output"
+AccuracyResult = namedtuple("AccuracyResult", ["position", "total_pages", "score"])
+
+def calculate_accuracy(responses_url: list[str], expected_url: str) -> AccuracyResult:
+    position: int = 0
+    total_pages: int = len(responses_url)
+    score: float = 0.0
+    expected_number = int(expected_url.split('/')[-2])
```
Comment (rngadam) [inline finesse/accuracy_functions.py:14]: regexp instead would make it clearer what you're extracting

**Review by rngadam — CHANGES_REQUESTED** [review]

File: finesse/FINESSE_USAGE.md (hunk 4 above)
Comment (ibrahim-kabir) [reply finesse/FINESSE_USAGE.md:1]: I've reviewed the documentation, and I believe it's fine as it is. The markdown file explains how the Finesse locust script works, and the README.md covers backend API testing, redirecting users to specific locust scripts for different purposes.

File: finesse/accuracy_functions.py (hunk 5 above)
Comment (ibrahim-kabir) [reply finesse/accuracy_functions.py]: Done

File: finesse/FINESSE_USAGE.md (hunk 3 above)
Comment (ibrahim-kabir) [reply finesse/FINESSE_USAGE.md]: Removed the picture and added the source of the diagram

**Comment by ibrahim-kabir** [comment]
> @ibrahim-kabir did you try with act locally ? image
> 
> Your'e calling workflow-lint-test-python but dont have a tests folder. Also your'e making use of workflow-markdown-check with a specified config file '.mlc_config.json' which is absent in your files.

Thx for your help, it works now

File: .github/workflows/python-api-test-workflows.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+name: Python api-test workflows
+
+on:
+  pull_request:
+    types:
+      - opened
+      - closed
+      - synchronize
+
+jobs:
```
Comment (rngadam) [inline .github/workflows/python-api-test-workflows.yml:10]: This has plenty of python code but does not include the Python lint and testing Github Workflows

Comment (rngadam) [reply .github/workflows/python-api-test-workflows.yml:10]: https://github.com/ai-cfia/github-workflows/blob/main/.github/workflows/workflow-lint-test-python.yml

**Review by rngadam — CHANGES_REQUESTED** [review]

File: .github/workflows/python-api-test-workflows.yml (hunk 6 above)
Comment (ibrahim-kabir) [reply .github/workflows/python-api-test-workflows.yml:10]: Done

**Comment by SonOfLope** [comment]
Merging with latest changes from main should allow the pipeline to pass. Related issue fixing this : https://github.com/ai-cfia/devops/pull/34

**Comment by rngadam** [comment]
> Merging with latest changes from main should allow the pipeline to pass. Related issue fixing this : [ai-cfia/devops#34](https://github.com/ai-cfia/devops/pull/34)

rebase on main right?

**Review by SonOfLope — APPROVED** [review]

**Review by rngadam — APPROVED** [review]
LGTM

**Pull request merged**